File: backend/src/DoctorLicenseManagement.Domain/Entities/Doctor.cs
using DoctorLicenseManagement.Domain.Enums;

namespace DoctorLicenseManagement.Domain.Entities;

public class Doctor
{
    public int Id { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Specialization { get; set; } = string.Empty;
    public string LicenseNumber { get; set; } = string.Empty;
    public DateTime LicenseExpiryDate { get; set; }
    public DoctorStatus Status { get; set; } = DoctorStatus.Active;
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
    public bool IsDeleted { get; set; }
}

File: web/features/doctors/types/doctor.ts
export type DoctorStatus = "Active" | "Expired" | "Suspended";

export type Doctor = {
  id: number;
  fullName: string;
  email: string;
  specialization: string;
  licenseNumber: string;
  licenseExpiryDate: string;
  status: DoctorStatus;
  createdDate: string;
};

export type DoctorFilters = {
  search?: string;
  status?: DoctorStatus | "";
};

export type CreateDoctorPayload = {
  fullName: string;
  email: string;
  specialization: string;
  licenseNumber: string;
  licenseExpiryDate: string;
  status: DoctorStatus;
};

export interface UpdateDoctorPayload extends CreateDoctorPayload {
  id: number;
}
